src/UI/SelectList.elm

```diff
@@ -0,0 +1,210 @@
+module UI.SelectList exposing (SelectList, selectList, toEl, withActionBar, withOptions, withSearchField)
+
+import Dict exposing (Dict)
+import Element exposing (Element, fill, shrink)
+import Element.Background as Background
+import Element.Border as Border
+import Element.Events as Events
+import Element.Font as Font
+import UI.ActionBar as ActionBar
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.Palette as Palette
+import UI.RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.TextField as TextField
+import UI.Utils.ARIA as ARIA
+import UI.Utils.Element as Element
+
+
+type alias SearchOptions object msg =
```

Comment:
We currently don't know how this use case will be, so I would remove it until we understand better what we need to do here.

Reply:
It's already working, on the spaces screen it got something like this, feedbacks are welcomed:
```elm
viewLoadedSpaces : List Space -> Dialog.State Dialog -> Maybe String -> RenderConfig -> Element Msg
viewLoadedSpaces spaces dialog currentFilter cfg =
    selectList (OpenSpaceDetails >> ForDialogMsg) .id spaceItemView
        |> SelectList.withOptions spaces
        |> SelectList.withSearchField
            "Space Name / Barcode"
            FilterSet
            (Maybe.map (\str -> ( str, spaceHasString )) currentFilter)
        |> SelectList.withActionBar "Create New Space"
            Icon.add
            (UuidGen (ForDialogMsg << OpenSpaceCreation))
        |> SelectList.toEl cfg

-- ...
spaceHasString : String -> Space -> Bool
spaceHasString str { name, barcode } =
    String.contains str name
        || String.contains str barcode
```

Reply:
I said that because we haven't defined if that would be a remote filter, local filter, and so on. it would probably be a remote filter as we will add pagination here. But I'm ok if you prefer to keep this.